### PR TITLE
Navigation update (top level flow visibility remix)

### PR DESCRIPTION
**Pull request by znicholasbrown**
PR Checklist:

- [x] add a short description of what's changed to the top of the `CHANGELOG.md`
- [x] add/update tests (or don't, for reasons explained below)

## Describe this PR

# ~~Opening this up for comments while it's a work in progress.~~ Ready 😄 

This PR reimagines the top-level app navigation by removing all major routes from the sidebar, placing them in places that make more sense. It replaces the current sidebar with a new, cleaner one that will allow 1) easy understanding of team-level directory segmentation of projects, flows, and tasks and 2) fast navigation between the 3, available from any part of the application.

Important changes:

***Refactored*** Navigation: most sidebar navigation routes have been moved to the top nav bar
  - The feedback button has been removed in favor of a combined "Feedback and Support" navigation item; the feedback button was outdated and redundant, given the support page
  - The `Home` route has been renamed to `Getting Started`, to better reflect its intended purpose
  - All `Team` pages are now harbored in the `Team` dropdown menu in the top nav
  - The `Calendar` and `Interactive API` are now top-level routes
  - *New* Added links to the UI Github and to the Slack community to the `Help` dropdown menu

***Visual update*** Connection menu: the connection menu now offers more information (and looks nicer!); the backend switcher (for moving between Cloud and Server installations) has been moved here for clarity

***Visual update*** User menu: the old user menu hadn't been touched in awhile so has been given some QOL improvements. In addition, all user settings have been moved here and are _not_ in the top level nav

***Visual update*** Global search bar: small changes here to reflect the new design of the nav bar in general, functionality unchanged

***New*** Directory sidebar: the old sidebar has been replaced with a new directory sidebar; from here you'll be able to navigate between projects, flows, and tasks with ease. You'll also be able to quickly switch between teams and immediately see the structure of your teams.
This was one of the primary motivators for this PR; we lack a way to imprecisely navigate between flows and projects, without first navigating to the dashboard page / flow tab. (#196)

***New*** Vuex store: `data`, which contains all projects and active flows (kept up-to-date from the app component) and tasks on an on-demand basis (but cached)
These are intended to remove the burden of querying for primary tables on individual components, keeping each in an up-to-date and updatable in-memory location instead

TODO:

- [x] Design a better system to keep the `active*` stores up to date
- [x] Fix fontawesome icon sizing on various pages (this replaces the webfont icons we've been using with svg icons instead for better design control)
- [x] Implement "active directory" functionality in the sidebar based on the `active*` stores
- [x] Re-introduce the team switcher to the new sidebar (removed for testing atm)
- [ ] Add search functionality to the directory sidebar (also figure out how best to implement this with on-demand data)
- [x] Write unit tests for new stores
- [x] Mobile functionality - the new top nav hasn't been properly tuned for mobile, so we'll need to collapse some menus on smaller screens

Resolves: #196

### CODE REVIEW

File: tests/unit/store/data.spec.js [hunk 1]
```diff
@@ -0,0 +1,673 @@
...
+      dataObjects.forEach(d => {
+        const active = 'active' + d.charAt(0).toUpperCase() + d.slice(1)
+        const plural = d + 's'
+        const id = active + 'Id'
+
+        test(`${active} getter should return the state activeFlow object`, () => {
+          expect(store.getters[active]).toEqual(state[active])
+        })
+
+        test(`${plural} getter should return a list of flows`, () => {
```
Comment (zhen0) [inline tests/unit/store/data.spec.js:175]: Just flows?  Looks like this is testing flows, projects and tasks?

**Review by zhen0 — COMMENTED** [review]
This is looking great @znicholasbrown.  The new navigation makes getting to flows and tasks much easier.  I also really like the more sensible placement of user settings.  A few small comments/questions from the code - the main comment is the backend switch with team and user pages - I can still click into and see my cloud teams if I switch to server while in the user section.  Not a major issue but I think a slightly confusing user experience. 

Screen Shot 2020-12-31 at 2 25 14 PM

File: src/components/LabelEdit.vue [hunk 2]
```diff
@@ -317,13 +317,12 @@ export default {
           <v-btn
             class="mt-0"
             icon
-            :disabled="labelResetDisabled"
```
[the remark this answers is not in the document]
Comment (znicholasbrown) [reply src/components/LabelEdit.vue:320]: Hm no, that might have been an accident when i was updating some of the font awesome icon refs, will revert that!

File: src/components/Nav/ConnectionMenu.vue [hunk 3]
```diff
@@ -0,0 +1,312 @@
...
+  'billing',
+  'welcome',
+  'current-plan',
+  'payment',
+  'api',
+  'name-team'
+]
+
+const backendProtectedRoutes = [
+  'team',
```
[the remark this answers is not in the document]
Comment (znicholasbrown) [reply src/components/Nav/ConnectionMenu.vue:24]: Hm after looking at this more closely, I've decided it's simpler to always route to the welcome screens (if cloud and a new team) or the dashboard

File: tests/unit/store/data.spec.js (hunk 1 above)
Comment (znicholasbrown) [reply tests/unit/store/data.spec.js]: Oops nope! I added the interpolated variables there, I wrote the non-looped test for flows and forgot to update the description.. fixed now! 😄

File: src/components/Nav/TeamSideNavButton.vue [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+  <v-btn
+    class="text-subtitle-1 text-capitalize mx-1 font-weight-medium d-flex justify-space-between"
+    dark
+    large
+    depressed
+    :width="size"
+    style="
+      background-color: rgba(255, 255, 255, 0.1);
+      transition: width 200ms;"
+    title="Open the team sidebar"
```
Comment (zhen0) [inline src/components/Nav/TeamSideNavButton.vue:32]: I'm not sure the team reference is true anymore/fully reflects what the sidenav shows 
```suggestion
    title="Open the sidebar"
```

**Review by zhen0 — COMMENTED** [review]
One very small suggestion for the sidebar tooltip, otherwise this LGTM @znicholasbrown

**Review by zhen0 — APPROVED** [review]
Thanks for the changes @znicholasbrown - LGTM!

**Pull request merged**